wagtail/core/models.py

```diff
@@ -2049,6 +2050,29 @@ def copy(self, recursive=False, to=None, update_attrs=None, copy_revisions=True,
                 page_copy.numchild = numchild
                 page_copy.save(clean=False, update_fields=['numchild'])
 
+            # Copy privacy permission
+            if self.get_view_restrictions().exists() and to:
+
+                # Issue 6212: The privacy settings are on the page being copied.
+                # On copy:
+                # 1.New parent has privacy settings. Do nothing
+                # 2. New parent has no privacy settings. Copy the settings (that is, create a new
+                # PageViewRestriction with the page foreign key set to the newly created page)
+                # The privacy settings are  on the page parent. On copy: do nothing.
+                # If the new parent has privacy settings, they will be inherited.
+
+                parent_page_restriction = to.get_view_restrictions()
+                if not parent_page_restriction.exists():
+                    privacy_settings_to_copy = get_object_or_404(PageViewRestriction, page_id=self.id)
```

Comment:
`get_object_or_404` would not be correct here - model code doesn't know anything about HTTP requests/responses, so it shouldn't be responsible for deciding whether to return a 404. However, this isn't really necessary anyway, since we're checking for the existence of a view restriction above - we can combine "does a view restriction exist?" with "retrieve the view restriction".

Overall, the new logic would look something like this:

    if to:
        parent_page_restriction = to.get_view_restrictions()
    else:
        parent_page_restriction = self.get_parent().get_view_restrictions()

    if not parent_page_restriction.exists():
        for privacy_settings_to_copy in self.view_restrictions.all():
            # copy the setting, as before